Runtime/UI/SafeArea.cs

```diff
@@ -0,0 +1,95 @@
+using UnityEngine;
+
+namespace Graphy.Runtime.UI
+{
+    /// <summary>
+    ///     Component that matches the size of the RectTransform to the safe area.
+    /// </summary>
+    [ExecuteAlways]
+    public sealed class SafeArea : MonoBehaviour
+    {
+        private Canvas m_canvas;
+        private bool m_isInitialized;
+        private Rect m_latestSafeArea;
+        private RectTransform m_rectTransform;
+        private DrivenRectTransformTracker m_rectTransformTracker;
+
+        private RectTransform RectTransform
+        {
+            get
+            {
+                if (m_rectTransform == null)
+                    m_rectTransform = (RectTransform)transform;
+
+                return m_rectTransform;
+            }
+        }
+
+        private Canvas Canvas
+        {
+            get
+            {
+                if (m_canvas == null)
+                    m_canvas = GetComponentInParent<Canvas>();
```

Comment:
Same, extract this into Awake or even better, just make it a [SerializedField] and we manually assign it in the prefab. GetComponentInParent is not that bad, but we wanna make sure Graphy is not affecting the game's performance if we can avoid it

Reply:
I moved this to `Awake()` as well. In EditMode, I want to use it immediately after attaching it, so I don't want to serialize it manually. But `GetComponentInParent()` will be called only once when it is attached in Edit Mode so no performance problems will occur.